### ISSUE TITLE
unexpected negation when no space between option letter and integer parameter

### ISSUE THREAD

**Issue by bentorkington** [tj/commander.js#1455]
the parser seems to negate an integer parameter whenever
* a program command is added, and
* and option to it is called without a space

```js
program
.command('update [directory...]')
.option('-j, --jobs <max>', 'Limit concurrent downloads. Default 2', 2)
.action(async (directory, opts) => {
  console.log(`number of jobs to run: ${opts.jobs}`);
});
program.parse(process.argv);
```

Output:
```
$ ./commandertest.js update -j 4
number of jobs to run: 4
$ ./commandertest.js update -j4 
number of jobs to run: -4    ***** EXPECTED: 4 *****
```

If the `command` line is removed, this reversing of sign does not happen.

**Comment by shadowspawn** [comment]
What version of commander are you using? I am unable to reproduce this with commander v7 and your example program.

(Note: if you have an async action handler, you should be calling `.parseAsync()` rather than `.parse()`. Not an issue for this program!)

**Comment by shadowspawn** [comment]
Ah, I am guessing you are using a version below v5.  Earlier versions did not have support for combining the option and option value into a single argument for a subcommand. Instead the short options were unconditionally expanded to separate short flags.

`-j4` --> `-j -4`

Fixed in #1145

**Comment by bentorkington** [comment]
Apologies, I thought I had updated but I was still stuck on an older version. Sorry for the invalid report, and thanks for the fast response.
